src/main.cpp

```diff
@@ -593,65 +596,31 @@ int main ( int argc, char** argv )
-            }
-        }
-        else
-        {
-            // the inifile is not supported for the headless server mode
-            if ( !strIniFileName.isEmpty() )
-            {
-                qWarning() << "No initialization file support in headless server mode.";
-            }
-        }
-
-        if ( strDirectoryServer.isEmpty() )
-        {
-            // per definition, we must be a headless server and ignoring inifile, so we have all the information
+            // by definition, when running with the GUI we always default to registering somewhere but
+            // until the settings are loaded we do not know where, so we cannot be prescriptive here
 
             if ( !strServerListFileName.isEmpty() )
             {
-                qWarning() << "Server list persistence file will only take effect when running as a directory server.";
-                strServerListFileName = "";
+                qInfo() << "Note:"
+                        << "Server list persistence file will only take effect when running as a directory server.";
             }
 
             if ( !strServerListFilter.isEmpty() )
             {
-                qWarning() << "Server list filter will only take effect when running as a directory server.";
-                strServerListFilter = "";
-            }
-
-            if ( !strServerPublicIP.isEmpty() )
-            {
-                qWarning() << "Server Public IP will only take effect when registering a server with a directory server.";
-                strServerPublicIP = "";
+                qInfo() << "Note:"
+                        << "Server list filter will only take effect when running as a directory server.";
             }
         }
         else
         {
-            // either we are not headless and there is an inifile, or a directory server was supplied on the command line
-
-            // if we are not headless, certain checks cannot be made, as the inifile state is not yet known
-            if ( !bUseGUI && strDirectoryServer.compare ( "localhost", Qt::CaseInsensitive ) != 0 && strDirectoryServer.compare ( "127.0.0.1" ) != 0 )
+            // the inifile is not supported for the headless server mode
+            if ( !strIniFileName.isEmpty() )
             {
-                if ( !strServerListFileName.isEmpty() )
-                {
-                    qWarning() << "Server list persistence file will only take effect when running as a directory server.";
-                    strServerListFileName = "";
-                }
-
-                if ( !strServerListFilter.isEmpty() )
-                {
-                    qWarning() << "Server list filter will only take effect when running as a directory server.";
-                    strServerListFileName = "";
-                }
+                qWarning() << "No initialization file support in headless server mode.";
```

Comment:
Maybe add `strIniFileName = "";` similarly to the other options, for consistency? I know the original code that was moved here didn't do it.

Reply:
Done